src/fondant/executor.py

```diff
@@ -254,7 +268,32 @@ def _execute_component(
         Returns:
             A `dd.DataFrame` instance with initial data.
         """
-        return component.load()
+        dask_df = component.load()
+
+        if self.index_column is None:
+            # Set monotonically increasing index
+            logger.info(
+                "Index column not specified, setting an automatic"
+                " monotonically increasing index",
+            )
+
+            def _set_unique_index(dataframe: pd.DataFrame, partition_info=None):
+                """Function that sets a unique index based on the partition and row number."""
+                dataframe["id"] = 1
+                dataframe["id"] = (
+                    str(partition_info["number"])
+                    + "_"
+                    + (dataframe.id.cumsum()).astype(str)
+                )
+                dataframe.index = dataframe.pop("id")
+                return dataframe
+
+            dask_df = dask_df.map_partitions(_set_unique_index, meta=dask_df.head())
+        else:
+            logger.info(f"Setting `{self.index_column}` as index")
+            dask_df = dask_df.set_index(self.index_column, drop=True)
```

Comment:
That might make sense, but I feel like we're adding a lot of arguments like this which increase the complexity. Since we only want to add this for one component at the moment, it might make more sense to just add it there and check if / how we want to generalize in the future.

Reply:
alright that makes sense, made the changes